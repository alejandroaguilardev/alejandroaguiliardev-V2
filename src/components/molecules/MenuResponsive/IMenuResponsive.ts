export interface IMenuResponsive {
    pages: string[];
    menu: boolean;
}