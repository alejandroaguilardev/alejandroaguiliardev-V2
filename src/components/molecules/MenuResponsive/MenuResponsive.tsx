import Link from "next/link"
import { Grid, Collapse } from "@mui/material"
import { Curriculum, PageRoute } from "../../atoms"
import { IMenuResponsive } from "./IMenuResponsive"

export const MenuResponsive = ({ pages, menu }: IMenuResponsive) => {
    return (
        <Collapse in={menu} sx={{ flexDirection: 'column', display: { xs: 'flex', md: 'none' } }} >
            <Grid sx={{ mt: 2, mb: 1 }}>
                {pages.map((page: string) => (
                    <PageRoute 
                        page={page} 
                        key={page}
                        sx={{ 
                            p: 3, 
                            borderTop: '1px  solid #999' 
                        }}
                    />
                ))}
                <Curriculum fullWidth={true} />
            </Grid>
        </Collapse >
    )
}
