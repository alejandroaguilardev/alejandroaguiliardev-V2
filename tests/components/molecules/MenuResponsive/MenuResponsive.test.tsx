import { render } from "@testing-library/react";
import { MenuResponsive } from "../../../../src/components/molecules/MenuResponsive/MenuResponsive";

jest.mock("../../../../src/components/atoms/PageRoute/PageRoute");
jest.mock("../../../../src/components/atoms/Mode/Mode");

describe("MenuResponsive Test", () => {
	const pages = ["page 1", "page 2"];
	const menu = true;

	test("toMacthSnapshot", () => {
		const { container } = render(<MenuResponsive pages={pages} menu={menu} />);

		expect(container).toMatchSnapshot();
	});
});
