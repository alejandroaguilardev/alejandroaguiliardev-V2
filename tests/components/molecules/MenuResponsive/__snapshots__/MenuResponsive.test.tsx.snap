// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuResponsive Test toMacthSnapshot 1`] = `
<div>
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-wi23bc-MuiCollapse-root"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiGrid-root css-1j400y7-MuiGrid-root"
        >
          <a
            href="https://drive.google.com/file/d/13xkWuHrDCPpXNNkmPCKqC0BOH6n9nxmo/view?usp=sharing"
            rel="”noopener noreferrer"
            target="_blank"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Get CV
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
